apps/ewallet/lib/ewallet/validators/transaction_consumption_validator.ex

```diff
@@ -102,6 +97,28 @@ defmodule EWallet.TransactionConsumptionValidator do
     end
   end
 
+  defp validate_not_expired(consumption) do
+    case TransactionConsumption.expired?(consumption) do
+      false ->
+        {:ok, consumption}
+
+      true ->
+        Event.dispatch(:transaction_consumption_finalized, %{consumption: consumption})
+        {:error, :expired_transaction_consumption}
+    end
+  end
+
+  defp validate_not_cancelled(consumption) do
+    case TransactionConsumption.cancelled?(consumption) do
+      false ->
+        {:ok, consumption}
+
+      true ->
+        Event.dispatch(:transaction_consumption_finalized, %{consumption: consumption})
```

Comment:
Why does it need to dispatch an event every time the validation failed?

Reply:
So in a case where `A` creates a transaction `request` with validation and `B` consumes it:
- `A` creates the request
- `B` consumes
- `A` receives a socket indicating that `B` wants to consume, `A` can then approve or reject this consumption
- `B` realized that he did something wrong and cancel his consumption
- `A` receives a socket indicating that the consumption is now in a `cancel` state
- `A` doesn't care and tries to approve it anyway (note that in a perfect world this shouldn't happen because a smart developer will prevent this when receiving the previous event. This is when it comes in this function and try to validate his confirmation).
- Confirmation fails because the consumption is cancelled. -> `B` is notified about it with the sockets.